### PR TITLE
Weather service api

### PR DESCRIPTION
**Pull request by Nepsoul**
#15 Implemented sunrise and sunset time display functionality in the weather information section, enhancing the user experience with accurate local time information for each country.

# Changes Made
- Added time conversion utility for Unix timestamps
- Implemented timezone-aware time display in CountryDetail component
- Enhanced weather API service to handle timezone data
- Added new time display components with proper formatting
- Updated weather state management to include sunrise/sunset data

# Configuration Changes
- Updated weather API service to include timezone information
- Modified CountryDetail component to handle time display
- -  
Testing
[x] Unit tests for time conversion functions
[x] Manual testing of time display across different countries
[x] Cross-browser testing (Chrome, Firefox, Safari)
[x] Mobile responsiveness testing
[x] Error handling testing for invalid timestamps

# Additional Notes
- Time display is automatically adjusted based on the country's timezone
- Added fallback display for cases where time data is unavailable
- Implemented proper error handling for timezone conversion
- Added loading states for time data fetching
- Time format follows the country's local time format
- - 
# Breaking Changes
None. The changes are backward compatible and don't affect existing functionality.

# Implementation Details
- Time Conversion
- Weather State Update
- Display Component

# Testing Plan
Unit Tests:
Time conversion function accuracy
Timezone handling
Error cases for invalid timestamps
Integration Tests:
Weather API response handling
Component rendering with time data
Error state handling
Manual Testing:
Different countries with various timezones
Edge cases (midnight, noon)
Different date formats
Error scenarios
Performance Testing:
Time conversion performance
Component render performance
Memory usage

# Future Improvements
- Add 24-hour time format option
- Implement timezone selection
- Add day/night cycle visualization
- Include timezone offset display
- This PR enhances the weather application by providing users with accurate sunrise and sunset times for their selected
-  country, improving the overall user experience and information completeness.

### CODE REVIEW

**Review by su-de-sh — APPROVED** [review]

File: Weather-Or-Not/src/components/CountryDetail.jsx [hunk 1]
```diff
@@ -1,35 +1,103 @@
...
+          sunset: formatTime(weatherData.sys.sunset),
+          temperature: weatherData.main.temp,
+          weatherIcon: weatherData.weather[0].icon,
+          humidity: weatherData.main.humidity,
+          maxTemp: weatherData.main.temp_max,
+          minTemp: weatherData.main.temp_min,
+          wind: weatherData.wind.speed,
+        });
+      } catch (error) {
+        toast.error(`Error fetching weather data: ${error.message}`);
```
Comment (tajpuriya27) [inline Weather-Or-Not/src/components/CountryDetail.jsx:45]: Add comment if you are sure what will be the error message here. Consider showing whole error message to client if you are not sure.

File: Weather-Or-Not/src/services/weather_api.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    const city = country.capital[0];
+    const weatherResponse = await axios.get(
+      `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${api_key}`
+    );
+    return weatherResponse.data;
+  } catch (error) {
+    throw error;
+  }
+};
```
Comment (tajpuriya27) [inline Weather-Or-Not/src/services/weather_api.js:19]: Pass the required city name while calling this function and avoid the if-condition each time this function gets called.
